addons/draggablesprite2d/src/draggablesprite2d.gd

```diff
@@ -1,79 +1,121 @@
 @export var texture : Texture2D : 
-	set(x):
-		texture = x
+	set(value):
+		texture = value
 		sprite.texture = texture
-		## Update the default collider with the shape and size of the sprite, if it exists
-		if collider and collider.shape:
-			collider.shape.size = Vector2(texture.get_width(), texture.get_height())
+		# Update the default_collider with the shape and size of the sprite, if it exists
+		update_default_collider()
 ## Whether or not the sprite should return to it's starting position when released
 @export var return_to_origin := false
 ## Whether or not it should be possible to grab the sprite
 @export var grabbable := true
 
-func _ready():
-	## Check if we need to instantiate a collider 
-	var children = get_children()
-	var needs_collider = true
-	for i in children:
-		## If a collision shape is a child, we don't need a collider
-		if i is CollisionShape2D or i is CollisionPolygon2D:
-			needs_collider = false
+
+## The Sprite node that will be used to display the texture
+var sprite := Sprite2D.new()
+## The default collider. It is automatically created and updated to match the size of the sprite
+var default_collider : CollisionShape2D
+## Whether or not the sprite is currently grabbed
+var is_grabbed := false :
+	set(value):
+		is_grabbed = value
+		if is_grabbed:
+			grabbed.emit()
+		else:
+			released.emit()
+
+
+# Helps a bit to make the dragging less choppy
+var grabbed_offset := Vector2.ZERO
+# Store for the original position
+var origin := Vector2.ZERO
+# Mouse button pressed tracker, used to essentially replicate the behavior of 'is_action_just_released'
+var mb_pressed = false
+
+
+func _ready() -> void:
+	# Connect signals
+	child_entered_tree.connect(_on_child_entered_tree)
+	child_exiting_tree.connect(_on_child_exiting_tree)
+	input_event.connect(_on_input_event)
+
+	# Add the sprite to the node
+	add_child(sprite)
+
+	# Create the default collider
+	default_collider = CollisionShape2D.new()
+	default_collider.shape = RectangleShape2D.new()
+
+	update_default_collider()
 	
-	## Otherwise, create a new collider from the size and shape of the sprite
-	if needs_collider:
-		collider = CollisionShape2D.new()
-		collider.shape = RectangleShape2D.new()
-		collider.shape.size = Vector2(texture.get_width() if texture else 0, texture.get_height() if texture else 0)
-		add_child(collider)
+	add_child(default_collider)
+	# Hide the default_collider if there is a custom collider
+	if has_custom_collider():
+		default_collider.visible = false
 	
-	## Set the starting origin if necessary
+	# Set the starting origin if necessary
 	if return_to_origin:
 		origin = position
-	
-	add_child(sprite)
-	input_event.connect(_on_input_event)
 
-## Mouse button pressed tracker, used to essentially replicate the behavior of 'is_action_just_released'
-var mb_pressed = false
 
-func _process(delta):
+func _process(delta) -> void:
 	# If the left mouse button is down and the object is and can be grabbed, update it's position
```

Comment:
Please update this comment (and line 80) now that the `input_method` import has been introduced.